Assets/Scripts/Player/PlayerController.cs

```diff
@@ -66,10 +66,13 @@ private void HandlePlayerMovement()
 
         private void HandlePlayerRotation()
         {
-            // Rotate the player to look in the direction of mouse position.
+            // Rotate the player to look in the direction of the mouse position.
             var dir = Input.mousePosition - Camera.main.WorldToScreenPoint(playerView.transform.position);
             var angle = Mathf.Atan2(dir.y, dir.x) * Mathf.Rad2Deg;
-            playerView.transform.rotation = Quaternion.AngleAxis(angle - 90, Vector3.forward);
+
+            // Calculate the desired rotation based on the rotation speed and time.deltaTime
+            var desiredRotation = Quaternion.AngleAxis(angle - 90, Vector3.forward);
+            playerView.transform.rotation = Quaternion.RotateTowards(playerView.transform.rotation, desiredRotation, playerScriptableObject.rotationSpeed * Time.deltaTime);
```

Comment:
Seems flawed. from Quaternion.AngleAxis(angle - 90, Vector3.forward) we get the exact rotation for the player ship we need for any frame. SO we would want to set that rotation on every frame.
The player rotation speed depends on simply how fast you move the mouse.